Index: connman-1.4/src/device.c
===================================================================
--- connman-1.4.orig/src/device.c
+++ connman-1.4/src/device.c
@@ -1247,7 +1247,10 @@ int __connman_device_init(const char *de
 	if (nodevice != NULL)
 		nodevice_filter = g_strsplit(nodevice, ",", -1);
 
+        /* Disable cleanup devices for machines booting via NFS */
+	#if 0        
 	cleanup_devices();
+	#endif 
 
 	return 0;
 }
