Index: SGMLSpm/sgmlspl.pl
===================================================================
--- SGMLSpm.orig/sgmlspl.pl	2010-04-30 11:34:37.000000000 -0700
+++ SGMLSpm/sgmlspl.pl	2010-04-30 11:34:38.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 ########################################################################
 # sgmlspl: a simple SGML postprocesser for the SGMLS and NSGMLS
 #          parsers (requires SGMLS.pm library).
Index: SGMLSpm/sgmlspl
===================================================================
--- SGMLSpm.orig/sgmlspl	2010-04-30 11:37:32.000000000 -0700
+++ SGMLSpm/sgmlspl	2010-04-30 11:38:01.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 ########################################################################
 # sgmlspl: a simple SGML postprocesser for the SGMLS and NSGMLS
 #          parsers (requires SGMLS.pm library).
