From 3676ed4021e7ceedb2e06c801910b919a0d732c4 Mon Sep 17 00:00:00 2001
From: Christopher Larson <chris_larson@mentor.com>
Date: Mon, 5 Dec 2016 20:47:42 -0700
Subject: [PATCH] site.py: obey sys.lib for sitepackages

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Christopher Larson <chris_larson@mentor.com>
---
 Lib/site.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

Index: Python-3.7.2/Lib/site.py
===================================================================
--- Python-3.7.2.orig/Lib/site.py
+++ Python-3.7.2/Lib/site.py
@@ -337,6 +337,10 @@ def getsitepackages(prefixes=None):
             sitepackages.append(os.path.join(prefix, sys.lib,
                                         "python%d.%d" % sys.version_info[:2],
                                         "site-packages"))
+            if sys.lib != "lib":
+                sitepackages.append(os.path.join(prefix, "lib",
+                                            "python" + sys.version[:3],
+                                            "site-packages"))
         else:
             sitepackages.append(prefix)
             sitepackages.append(os.path.join(prefix, sys.lib, "site-packages"))
