--- orig/ust-0.12/include/ust/clock.h	2011-02-21 18:14:02.000000000 +0500
+++ ust-0.12/include/ust/clock.h	2011-07-28 15:44:14.279814053 +0500
@@ -76,7 +76,7 @@
 	return retval;
 }
 
-#if __i386__ || __x86_64__
+#if __i386__ || __x86_64__ || __powerpc__ || __mips__
 static __inline__ uint64_t trace_clock_frequency(void)
 {
 	struct timespec ts;
@@ -88,12 +88,12 @@
 	}
 	return 1000000000LL;
 }
-#else /* #if __i386__ || __x86_64__ */
+#else /* #if __i386__ || __x86_64__ || __powerpc__ || __mips__ */
 static __inline__ uint64_t trace_clock_frequency(void)
 {
 	return 1000000000LL;
 }
-#endif /* #else #if __i386__ || __x86_64__ */
+#endif /* #else #if __i386__ || __x86_64__ || __powerpc__ || __mips__ */
 
 static __inline__ uint32_t trace_clock_freq_scale(void)
 {
