From bb65dc2fb6507dd4925d87b3cf54bd459821a290 Mon Sep 17 00:00:00 2001
From: Christopher Larson <chris_larson@mentor.com>
Date: Wed, 30 Nov 2016 09:51:05 -0700
Subject: [PATCH 3/3] Also define amd64_linux funcs for x32

Signed-off-by: Christopher Larson <chris_larson@mentor.com>
---
 gdb/gdbserver/linux-x86-tdesc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdb/gdbserver/linux-x86-tdesc.h b/gdb/gdbserver/linux-x86-tdesc.h
index 720f50c..126292f 100644
--- a/gdb/gdbserver/linux-x86-tdesc.h
+++ b/gdb/gdbserver/linux-x86-tdesc.h
@@ -33,7 +33,7 @@ enum x86_linux_tdesc {
 
 #ifdef __x86_64__
 
-#if defined __LP64__  || !defined IN_PROCESS_AGENT
+#if defined __LP64__  || defined __ILP32__ || !defined IN_PROCESS_AGENT
 /* Defined in auto-generated file amd64-linux.c.  */
 void init_registers_amd64_linux (void);
 extern const struct target_desc *tdesc_amd64_linux;
-- 
2.8.0

