From 3be46ea060a153b9d02a0e141ce9f06e267a724a Mon Sep 17 00:00:00 2001
From: Srikanth Krishnakar <Srikanth_Krishnakar@mentor.com>
Date: Tue, 22 Mar 2016 19:33:47 +0530
Subject: [PATCH] fs/Makefile: Re-order EXT4 prior to EXT2 or EXT3

Signed-off-by: Srikanth Krishnakar <Srikanth_Krishnakar@mentor.com>
---
 fs/Makefile | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/fs/Makefile b/fs/Makefile
index cb92fd4..bb47469 100644
--- a/fs/Makefile
+++ b/fs/Makefile
@@ -62,11 +62,9 @@ obj-$(CONFIG_DLM)		+= dlm/
 # Do not add any filesystems before this line
 obj-$(CONFIG_FSCACHE)		+= fscache/
 obj-$(CONFIG_REISERFS_FS)	+= reiserfs/
+obj-$(CONFIG_EXT4_FS)		+= ext4/
 obj-$(CONFIG_EXT3_FS)		+= ext3/ # Before ext2 so root fs can be ext3
 obj-$(CONFIG_EXT2_FS)		+= ext2/
-# We place ext4 after ext2 so plain ext2 root fs's are mounted using ext2
-# unless explicitly requested by rootfstype
-obj-$(CONFIG_EXT4_FS)		+= ext4/
 obj-$(CONFIG_JBD)		+= jbd/
 obj-$(CONFIG_JBD2)		+= jbd2/
 obj-$(CONFIG_CRAMFS)		+= cramfs/
-- 
1.9.1

