From 5dc4fd6245a837bb367600ab4f478a3f852e5cc5 Mon Sep 17 00:00:00 2001
From: Srikanth Krishnakar <Srikanth_Krishnakar@mentor.com>
Date: Fri, 30 Sep 2016 13:51:02 +0530
Subject: [PATCH] psplash: disable progress-bar for systemd

Signed-off-by: Srikanth Krishnakar <Srikanth_Krishnakar@mentor.com>
---
 psplash.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

Index: git/psplash.c
===================================================================
--- git.orig/psplash.c
+++ git/psplash.c
@@ -107,10 +107,12 @@ parse_command (PSplashFB *fb, char *stri
 
   if (!strcmp(command,"PROGRESS")) 
     {
-      char *arg = strtok(NULL, "\0");
+       /* Do not draw any progress in case of systemd.
+       char *arg = strtok(NULL, "\0");
 
       if (arg)
         psplash_draw_progress (fb, atoi(arg));
+        */
     } 
   else if (!strcmp(command,"MSG")) 
     {
@@ -304,6 +306,7 @@ main (int argc, char** argv)
 			 POKY_IMG_ROWSTRIDE,
 			 POKY_IMG_RLE_PIXEL_DATA);
 
+#if 0
   /* Draw progress bar border */
   psplash_fb_draw_image (fb, 
 			 (fb->width  - BAR_IMG_WIDTH)/2, 
@@ -319,6 +322,7 @@ main (int argc, char** argv)
 #ifdef PSPLASH_STARTUP_MSG
   psplash_draw_msg (fb, PSPLASH_STARTUP_MSG);
 #endif
+#endif /* Disable progress bar for systemd*/
 
   /* Scene set so let's flip the buffers. */
   /* The first time we also synchronize the buffers so we can build on an
