diff -Naur old/rsync.h new/rsync.h
--- old/rsync.h	2014-04-08 18:21:34.000000000 +1100
+++ new/rsync.h	2014-04-15 20:28:02.000000000 +1100
@@ -82,7 +82,7 @@
 			     == ((unsigned)(b2) & (unsigned)(mask)))
 
 /* update this if you make incompatible changes */
-#define PROTOCOL_VERSION 30
+#define PROTOCOL_VERSION 29
 
 /* We refuse to interoperate with versions that are not in this range.
  * Note that we assume we'll work with later versions: the onus is on
