diff -Naur old/tmpfiles.d/legacy.conf new/tmpfiles.d/legacy.conf
--- old/tmpfiles.d/legacy.conf	2013-01-07 15:50:49.000000000 -0600
+++ new/tmpfiles.d/legacy.conf	2013-06-19 00:38:06.000000000 -0500
@@ -15,7 +15,7 @@
 # /run/lock/subsys is used for serializing SysV service execution, and
 # hence without use on SysV-less systems.
 
-d /run/lock/subsys 0755 root root -
+#d /run/lock/subsys 0755 root root -
 
 # /run/lock/lockdev is used to serialize access to tty devices via
 # LCK..xxx style lock files, For more information see:
@@ -23,7 +23,7 @@
 # On modern systems a BSD file lock is a better choice if
 # serialization is needed on those devices.
 
-d /run/lock/lockdev 0775 root lock -
+#d /run/lock/lockdev 0775 root lock -
 
 # /forcefsck, /fastboot and /forcequotecheck are deprecated in favor of the
 # kernel command line options 'fsck.mode=force', 'fsck.mode=skip' and
