From 2098470a38e16173c23d176fe87c64754c2000e5 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Tue, 9 Jun 2020 21:48:36 +0200
Subject: [PATCH] fix ordering cycle with private tmp

Upstream-Status: Backport
---
 init.d/service.fedora | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/init.d/service.fedora b/init.d/service.fedora
index b8f32ea..8de8828 100644
--- a/init.d/service.fedora
+++ b/init.d/service.fedora
@@ -12,7 +12,8 @@ SuccessExitStatus=137 143
 
 SecureBits=noroot-locked
 CapabilityBoundingSet=CAP_SYS_ADMIN
-PrivateTmp=true
+# We can *not* set PrivateTmp=true as it can cause an ordering cycle.
+PrivateTmp=false
 PrivateDevices=true
 PrivateNetwork=true
 ProtectSystem=full
-- 
2.17.1

