Index: git/session.cxx
===================================================================
--- git.orig/session.cxx
+++ git/session.cxx
@@ -1634,7 +1634,7 @@ systemtap_session::parse_kernel_function
	clog << _F("Kernel symbol table %s unavailable, (%s)",
		   system_map_path.c_str(), strerror(errno)) << endl;
 
-      string system_map_path2 = "/boot/System.map-" + kernel_release;
+      string system_map_path2 = kernel_build_tree + "/System.map-" + kernel_release;
       system_map.clear();
       system_map.open(system_map_path2.c_str(), ifstream::in);
       if (! system_map.is_open())
