---
 configure.in |    1 +
 1 file changed, 1 insertion(+)

--- valgrind-3.7.0.orig/configure.in
+++ valgrind-3.7.0/configure.in
@@ -111,6 +111,7 @@ AC_MSG_CHECKING([for a supported version
              | $SED 's/i686-apple-darwin10//' \
              | $SED 's/i686-apple-darwin11//' \
              | $SED 's/.*GCC[^0-9]*//' \
+             | $SED 's/.*)//' \
              | $SED 's/^[^0-9]*\([0-9.]*\).*$/\1/'`]
 
 is_clang="notclang"
