diff -Naur old/configure.ac new/configure.ac
--- old/configure.ac	2014-03-25 01:53:26.000000000 -0500
+++ new/configure.ac	2014-03-25 01:55:24.699469386 -0500
@@ -86,7 +86,9 @@
 PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.30.0])
 PKG_CHECK_MODULES([GIO], [gio-2.0 >= 2.30.0])
 PKG_CHECK_MODULES([GIO_UNIX], [gio-unix-2.0 >= 2.30.0])
-PKG_CHECK_MODULES([SYSTEMD_DAEMON], [libsystemd-daemon >= 183])
+PKG_CHECK_MODULES([SYSTEMD_DAEMON], [libsystemd-daemon >= 183],, [
+    PKG_CHECK_MODULES([SYSTEMD_DAEMON], [libsystemd >= 209])
+])
 PKG_CHECK_MODULES([DLT], [automotive-dlt >= 2.2.0])
 
 dnl *********************************************
