---
 Makefile |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- git.orig/Makefile
+++ git/Makefile
@@ -24,12 +24,13 @@ subdirs:
	done && test -z "$$fail"

 chkconfig: $(OBJS)
-	$(CC) $(LDFLAGS) -o chkconfig $(OBJS) -lpopt
+	$(CC) $(LDFLAGS) -o chkconfig $(OBJS) $(LIBS) -lpopt

 alternatives: alternatives.o
+	$(CC) $(LDFLAGS) -o alternatives alternatives.o $(LIBS)

 ntsysv: $(NTOBJS)
-	$(CC) $(LDFLAGS) -o ntsysv $(NTOBJS) -lnewt -lpopt $(LIBMHACK)
+	$(CC) $(LDFLAGS) -o ntsysv $(NTOBJS) -lnewt -lpopt $(LIBS) $(LIBMHACK)

 chkconfig.o: chkconfig.c leveldb.h
	$(CC) $(CFLAGS) -DVERSION=\"$(VERSION)\" -c chkconfig.c
